src/js/components/CommandPanel/CommandPanel.tsx

```diff
@@ -0,0 +1,723 @@
+                                <Button
+                                    variant="contained"
+                                    color="primary"
+                                    disableElevation
+                                    onClick={handleSaveEditedCommandClick}
+                                >
+                                    Save
+                                </Button>
+                                <Button
+                                    variant="outlined"
+                                    color="primary"
+                                    disableElevation
+                                    onClick={handleCancelEditingCommandClick}
+                                >
+                                    Cancel
+                                </Button>
+                            </>
+                        )
+                    }
+
+                </Box>
+
+                {preparedCommands.length > 0 && (
+                    <>
+                        <Typography variant="h6" sx={{fontWeight: 400, display: 'flex', alignItems: 'center', gap: 1}}>
+                            Message command list
+                            {createCommandDirectionIcon(preparedCommands[0].command.value.directionType)}
+                        </Typography>
+
+                        <Box sx={{
+                            flex: '1 1 auto',
+                            minHeight: '300px',
+                            display: 'flex',
+                            flexDirection: 'column',
+                            gap: 2
+                        }}>
+                            <DragDropContext onDragEnd={onPreparedCommandDragEnd}>
+                                <Droppable droppableId="preparedCommandList">
+                                    {(provided) => (
+                                        <List
+                                            dense
+                                            ref={provided.innerRef}
+                                            {...provided.droppableProps}
+                                            sx={{
+                                                maxHeight: '100%',
+                                                overflow: 'auto',
+                                                p: 0,
+                                                borderWidth: '1px',
+                                                borderStyle: 'solid',
+                                                borderColor: 'divider'
+                                            }}
+                                        >
+                                            {preparedCommands.map((preparedCommand, index) => (
+                                                <Draggable key={preparedCommand.id} draggableId={preparedCommand.id} index={index}>
+                                                    {(provided) => (
+                                                        <ListItem
```

Comment:
isn't it too nested?

Reply:
Yes, the `CommandPanel` component needs to be divided into smaller components.

Discussed, will be implemented in the future.